internal/api/create_credentials_v1alpha1.go

```diff
@@ -14,7 +14,7 @@ import (
 	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
 )
 
-type credentials struct {
+type specificCredentials struct {
```

Comment:
I see this was renamed to differentiate it clearly from the new `genericCredentials` type, but I think we could leave this as it was (`credentials`) and rename `genericCredentials` to just `secret` -- which I think is cleaner because it doesn't imply any assumption that all secrets are credentials, which isn't necessarily true.